apps/scan/frontend/src/screens/poll_worker_screen.tsx

```diff
@@ -169,268 +332,191 @@ export function PollWorkerScreen({
-      </Screen>
-    );
-  }
-
-  if (pollWorkerFlowState === 'polls_transition_processing') {
-    assert(lastPollsTransition);
-    const pollsTransitionProcessingText = (() => {
-      const pollsTransitionType = lastPollsTransition.type;
-      switch (pollsTransitionType) {
-        case 'close_polls':
-          return 'Closing Polls…';
-        case 'open_polls':
-          return 'Opening Polls…';
-        case 'pause_voting':
-          return 'Pausing Voting…';
-        case 'resume_voting':
-          return 'Resuming Voting…';
-        /* istanbul ignore next - compile-time check for completeness */
-        default:
-          throwIllegalValue(pollsTransitionType);
-      }
-    })();
-
-    return (
-      <Screen>
-        <LoadingAnimation />
-        <CenteredLargeProse>
-          <H1>{pollsTransitionProcessingText}</H1>
-        </CenteredLargeProse>
-      </Screen>
-    );
-  }
-
-  if (
-    pollWorkerFlowState === 'polls_transition_complete' ||
-    pollWorkerFlowState === 'reprinting_previous_report_complete'
-  ) {
-    assert(lastPollsTransition);
-    const primaryText = (() => {
-      if (pollWorkerFlowState === 'reprinting_previous_report_complete') {
-        return '';
-      }
-
-      const pollsTransitionType = lastPollsTransition.type;
-      switch (pollsTransitionType) {
-        case 'close_polls':
-          return 'Polls are closed.';
-        case 'open_polls':
-          return 'Polls are open.';
-        case 'resume_voting':
-          return 'Voting resumed.';
-        case 'pause_voting':
-          return 'Voting paused.';
-        /* istanbul ignore next - compile-time check for completeness */
-        default:
-          throwIllegalValue(pollsTransitionType);
-      }
-    })();
+  const allowReprintingReport =
+    lastPollsTransition &&
+    lastPollsTransition.ballotCount === scannedBallotCount;
 
-    return (
-      <Screen>
-        <CenteredLargeProse>
-          {primaryText && <H1>{primaryText}</H1>}
-          <P>
-            Insert{' '}
-            {numReportPages
-              ? `${numReportPages} ${pluralize(
-                  'sheet',
-                  numReportPages
-                )} of paper`
-              : 'paper'}{' '}
-            into the printer to print the report.
-          </P>
-          <P>
-            Remove the poll worker card once you have printed all necessary
-            reports.
-          </P>
-          <P>
-            <ReprintReportButton
-              lastPollsTransition={lastPollsTransition}
-              scannedBallotCount={scannedBallotCount}
-              isAdditional
-              beforePrint={() => setPollWorkerFlowState('reprinting_report')}
-              afterPrint={() =>
-                setPollWorkerFlowState('polls_transition_complete')
-              }
-            />
-          </P>
-        </CenteredLargeProse>
-      </Screen>
-    );
+  if (isShowingBallotsAlreadyScannedScreen) {
+    return BallotsAlreadyScannedScreen;
```

Comment:
Wait, how does this work? Shouldn't it be `<BallotsAlreadyScannedScreen />`?

Reply:
Good catch! I guess it has been working, because it was the same before. React must accept / handle this, although perhaps not exactly as we expect. Will switch it to JSX syntax.

Reply:
Oh I just looked at the definition of `BallotsAlreadyScannedScreen` - it's a JSX value, not a component function, so that explains it.